hphp/runtime/vm/jit/vasm-lower.h

```diff
@@ -25,14 +25,28 @@ namespace HPHP { namespace jit {
 
 struct Vunit;
 
+/*
+ * Vasm-to-vasm lowering state.
+ */
+struct VLS {
+  Vunit& unit;
+  bool allow_vreg;
+};
+
+enum class VregConstraint : uint8_t {
```

Comment:
If we just use integers to denote the "restriction level", we can get rid of this.